From 057a93f68996e1fa58c3f79373aa84536c3c86aa Mon Sep 17 00:00:00 2001
From: Ye Li <ye.li@nxp.com>
Date: Wed, 14 Jul 2021 00:09:37 -0700
Subject: [PATCH 04/19] LFU-193-2 usb: xhci-imx8m: Update iMX8MP XHCI driver to
 support new DTS node

We will sync the USB DTS node with latest kernel DTS. This node has
two levels structure. The top level is used for hsiomix GPR register
access for USB wakeup which is not needed in u-boot. So we just
add a MISC driver to pass through to next DWC node to match the
XHCI driver

Signed-off-by: Ye Li <ye.li@nxp.com>
Reviewed-by: Peng Fan <peng.fan@nxp.com>
---
 drivers/usb/host/Kconfig      |  1 +
 drivers/usb/host/xhci-imx8m.c | 11 +++++++++++
 2 files changed, 12 insertions(+)

diff --git a/drivers/usb/host/Kconfig b/drivers/usb/host/Kconfig
index 7f4ac58cc4..4fdbf6212b 100644
--- a/drivers/usb/host/Kconfig
+++ b/drivers/usb/host/Kconfig
@@ -113,6 +113,7 @@ config USB_XHCI_IMX8
 config USB_XHCI_IMX8M
 	bool "XHCI support for imx8M(mscale)"
 	depends on ARCH_IMX8M
+	select MISC if IMX8MP
 	default y
 	help
 	  Enables support for the on-chip xHCI controller on imx8m(mscale) SoC.
diff --git a/drivers/usb/host/xhci-imx8m.c b/drivers/usb/host/xhci-imx8m.c
index 1681a3a715..5756a967ad 100644
--- a/drivers/usb/host/xhci-imx8m.c
+++ b/drivers/usb/host/xhci-imx8m.c
@@ -200,6 +200,17 @@ U_BOOT_DRIVER(xhci_imx8m) = {
 	.flags	= DM_FLAG_ALLOC_PRIV_DMA,
 };
 
+static const struct udevice_id xhci_imx8mp_ids[] = {
+	{ .compatible = "fsl,imx8mp-dwc3", },
+	{ }
+};
+
+U_BOOT_DRIVER(xhci_imx8mp_misc) = {
+	.name	= "xhci_imx8mp_misc",
+	.id	= UCLASS_MISC,
+	.of_match = of_match_ptr(xhci_imx8mp_ids),
+};
+
 #else
 int xhci_hcd_init(int index, struct xhci_hccr **hccr, struct xhci_hcor **hcor)
 {
-- 
2.33.0

